analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CollectionEmptinessChecking.Fixed.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Tests.Diagnostics
+{
+    public class CollectionEmptinessChecking
+    {
+        private static bool HasContent1(IEnumerable<string> l)
+        {
+            return l.Any(); // Fixed
+        }
+        private static bool HasContent1b(IEnumerable<string> l)
+        {
+            return l.Any(); // Fixed
+        }
+        private static bool HasContent2(List<string> l)
+        {
+            return Enumerable.Any(l); // Fixed
```

Comment:
Can you add a test case demonstrating that `!` operator will behave nicely with this case?

Reply:
I rewrote the tests, using naming to make all scenario's clear. Hop that it now clear which scenario's are covered.